hilti/include/rt/compiler-setup.h

```diff
@@ -5,6 +5,9 @@
 
 #pragma once
 
+// Our type info infrastructure needs this, and it seems to work for our cases.
+#pragma clang diagnostic ignored "-Winvalid-offsetof"
```

Comment:
This is a pretty big hammer. Any chance to move it closer to the affected code? Something like (i.e., `push` it before the suspect call, `pop` it again after).

Reply:
Not easy to move there as it's inside a block of `static ...` declarations, we'd need to add something to the formatter API to prefix/postfix. I'm not too worried, though, it's a pretty rare feature to use.